### ISSUE TITLE
Fit multiple background components from simulations

### ISSUE THREAD

**Issue by israelmcmc** [cositools/cosipy#203]

**Comment by israelmcmc** [comment]
For now, it would be fine to add the capabilities, in principle, to fit an arbitrary number of components. But work on making it converge for only 2. See also #207 for many many components.

**Comment by GallegoSav** [comment]
@israelmcmc @ckarwin  If I understood well the fit is not even converging when using the total background model right ?

**Comment by israelmcmc** [comment]
@GallegoSav Yes, that's correct. It converged for one single component, but not when summing up all of them. It's not clear to me if adding more background was what caused this, or if it's simply that the fit is not very robust and we just got lucky before. In any case, this would be the starting point indeed, but any trick to make it convert should also help the multiple components fit.

**Comment by israelmcmc** [comment]
@GallegoSav I was asking Yong to add the tests for this PR, but you'll need to do it as well at some point hehe. Here are some instructions: https://github.com/cositools/cosipy/blob/c8eb53631f686621d93a8b56395ce66ae3d71825/docs/dev/unit_tests.rst

**Comment by GallegoSav** [comment]
> @GallegoSav Yes, that's correct. It converged for one single component, but not when summing up all of them. It's not clear to me if adding more background was what caused this, or if it's simply that the fit is not very robust and we just got lucky before. In any case, this would be the starting point indeed, but any trick to make it convert should also help the multiple components fit.

@israelmcmc It seems to be due to high background level indeed I have similar error when I am using just the albedo photon scale by 10 or the total background .
![image](https://github.com/user-attachments/assets/bb15dfe4-148c-493a-8d77-91bae4ad011d)
![image](https://github.com/user-attachments/assets/b1cf707d-c51d-40e7-bd2f-077b117e1641)
![image](https://github.com/user-attachments/assets/2dbc370f-b257-4b5b-8699-7221d891227f)

**Comment by israelmcmc** [comment]
@GallegoSav I see. I'd check the significance of the Crab when using the total background, to see if this is below the sensitivity. Using the 3ML, you can get the log-likelihood with and without the source. Keep all parameters fixed to the known values and get the likelihood (source hypothesis L1). Then set the source flux normalization to 0 and get the likelihood again (null hypothesis L0). The TS = 2*log(L1/L0). I remember @ckarwin did this, but I don't remember exactly for which case.

**Comment by GallegoSav** [comment]
> @GallegoSav I see. I'd check the significance of the Crab when using the total background, to see if this is below the sensitivity. Using the 3ML, you can get the log-likelihood with and without the source. Keep all parameters fixed to the known values and get the likelihood (source hypothesis L1). Then set the source flux normalization to 0 and get the likelihood again (null hypothesis L0). The TS = 2*log(L1/L0). I remember @ckarwin did this, but I don't remember exactly for which case.

I obtained TS=1e-4 , assuming Wilks's Theorem and 4 degrees of freedom that would give pvalue close to one so no possibility to exclude the null hypothesis

**Comment by israelmcmc** [comment]
Thanks @GallegoSav. I'll open the discussion for this on Slack, since this doesn't seem like an issue with the code itself.

**Comment by GallegoSav** [comment]
I put the slack conversation here to keep track of the discussion : https://ucbssl.slack.com/archives/C02TWC7U4NL/p1721837271221179

**Comment by israelmcmc** [comment]
PR #369 should help with this. There's an interface called `BackgroundInterface`, which accepts an arbitrary number of parameters --e.g. the normalization of each background. There's currently only a single implementation `FreeNormBinnedBackground`, which handles a simple normalization value, reproducing our current results.

While this by itself doesn't solve this issue, it should now be easier to implement fitting multiple components, since the [`LikelihoodInterface`](https://github.com/israelmcmc/cosipy/blob/d1774859ca2d146556ba4690d9ab5ec505727643/cosipy/interfaces/likelihood_interface.py#L12) (which together with [`ThreeMLPluginInterface`](https://github.com/israelmcmc/cosipy/blob/d1774859ca2d146556ba4690d9ab5ec505727643/cosipy/interfaces/threeml_plugin_interface.py#L9C7-L9C29) replaces COSILike) can handle an arbitrary `BackgroundInterface`

**Comment by GallegoSav** [comment]
@israelmcmc Should we close this issue since the test we made during COSIfest was successful ?

**Comment by israelmcmc** [comment]
Yeah! It's technically not yet merged into develop, but it's imminent and it feels good to close issues and reduce the clutter :)
